lib/api_auth/base.rb

```diff
@@ -67,7 +69,7 @@ def generate_secret_key
 
     private
 
-    AUTH_HEADER_PATTERN = /APIAuth ([^:]+):(.+)$/
+    AUTH_HEADER_PATTERN = /APIAuth(-HMAC-(SHA[0-9]+))? ([^:]+):(.+)$/
```

Comment:
Not sure if it's worth it, but you could make the outer (-HMAC) part non-capturing since we're not using it for anything.